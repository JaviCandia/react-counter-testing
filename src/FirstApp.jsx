import PropTypes from "prop-types";

const nameList = () => {
  const names = ["Javier", "Maribel", "Chofis", "Wero"];
  return names.map((name, index) => <li key={index}> {name} </li>);
};

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h2 data-testid="test-subtitle"> {subtitle} </h2>
      <p>{name}</p>
      <p>Checa esta lista:</p>
      <ul>{nameList()}</ul>
    </>
  );
};

FirstApp.propTypes = {
  subtitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  name: "Fernando",
  subtitle: "No hay subtitulo",
  // title: "No hay titulo",
};
