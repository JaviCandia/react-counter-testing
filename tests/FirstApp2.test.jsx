import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("FirstApp", () => {
  const title = "Hello world!";
  const subtitle = "welcome!";

  it("should match snapshot", () => {
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  it("should show title in h1", () => {
    render(<FirstApp title={title} />);

    expect(screen.getByTestId("test-title").innerHTML).toContain(title);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(title);
  });

  it("should show subtitle", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);

    expect(screen.getByTestId("test-subtitle").innerHTML).toContain(subtitle);
    expect(screen.getByText(subtitle)).toBeTruthy();
  });
});
