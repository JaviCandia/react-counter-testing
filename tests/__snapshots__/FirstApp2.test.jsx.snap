// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp should match snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello world!
  </h1>
  <h2
    data-testid="test-subtitle"
  >
     
    No hay subtitulo
     
  </h2>
  <p>
    Fernando
  </p>
  <p>
    Checa esta lista:
  </p>
  <ul>
    <li>
       
      Javier
       
    </li>
    <li>
       
      Maribel
       
    </li>
    <li>
       
      Chofis
       
    </li>
    <li>
       
      Wero
       
    </li>
  </ul>
</div>
`;
